app/controllers/subjects_controller.rb

```diff
@@ -13,7 +13,15 @@ def show
   end
 
   def all
-    @subjects = TreePreloader.new.preload
+    @subjects =
+      begin
```

Comment:
instead of using begin, couldn't this be this way:?
```
subjects =
      if params[:search].present?
        Subject.where("lower(unaccent(name)) LIKE ?", "%#{I18n.transliterate(params[:search].downcase)}%")
      end

@subjects = TreePreloader.new(subjects).preload
   ````

Reply:
Updated in 7ab7e5f!